pandas/tests/io/formats/test_to_markdown.py

```diff
@@ -53,3 +54,36 @@ def test_no_buf(capsys):
     assert (
         result == "|    |   0 |\n|---:|----:|\n|  0 |   1 |\n|  1 |   2 |\n|  2 |   3 |"
     )
+
+
+@pytest.mark.parametrize("index", [True, False, None])
+@pytest.mark.parametrize("showindex", [True, False, None])
+def test_index(index, showindex):
+    # GH 32667
+    kwargs = {}
+    if index is not None:
+        kwargs["index"] = index
+    if showindex is not None:
+        kwargs["showindex"] = showindex
+
+    df = pd.DataFrame([1, 2, 3])
+    yes_index_result = (
+        "|    |   0 |\n|---:|----:|\n|  0 |   1 |\n|  1 |   2 |\n|  2 |   3 |"
+    )
+    no_index_result = "|   0 |\n|----:|\n|   1 |\n|   2 |\n|   3 |"
+
+    warning = FutureWarning if "showindex" in kwargs else None
+    with tm.assert_produces_warning(warning):
+        result = df.to_markdown(**kwargs)
+
+    if "showindex" in kwargs:
+        # give showindex higher priority if specified
+        if showindex:
+            assert result == yes_index_result
```

Comment:
nitpick, instead of 4 asserts, maybe 4 expected= and then 1 `assert result==expected` at the end.